### PR TITLE
[#167102226] Added migration of store for organization entitie

### PR DESCRIPTION
**Pull request by laurapugliese**
Added migration of store for organization entitie

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#167102226](https://www.pivotaltracker.com/story/show/167102226): Come CIT voglio visualizzare un componente che fornisca la lista degli enti che possono essere visualizzati nella sezione locali

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/boot/configureStoreAndPersistor.ts [hunk 1]
```diff
@@ -62,7 +62,17 @@ const migrations: MigrationManifest = {
...
-  }
+  },
+
+  // Version 3
+  // we changes the organizations so we clear the entitie of organizations to force a reload
+  "3": (state: PersistedState): PersistedState =>
+    ({
+      ...state,
+      entities: {
+        organizations: {}
```
Comment (fpersico) [inline ts/boot/configureStoreAndPersistor.ts]: Aren't you deleting all the other entities?

Comment (laurapugliese) [reply ts/boot/configureStoreAndPersistor.ts]: Fixed in commit 1565112

File: ts/boot/configureStoreAndPersistor.ts [hunk 2]
```diff
@@ -62,7 +62,18 @@ const migrations: MigrationManifest = {
...
-  }
+  },
+
+  // Version 3
+  // we changes the organizations so we clear the entitie of organizations to force a reload
+  "3": (state: PersistedState): PersistedState =>
+    ({
+      ...state,
+      entities: {
+        ...((state as any).entities ? (state as any).entities : {}),
```
Comment (fpersico) [inline ts/boot/configureStoreAndPersistor.ts]: No any please.

File: ts/boot/configureStoreAndPersistor.ts [hunk 3]
```diff
@@ -62,7 +62,18 @@ const migrations: MigrationManifest = {
...
+        organizations: {}
+      }
+    } as PersistedState)
```
Comment (fpersico) [inline ts/boot/configureStoreAndPersistor.ts]: It should not be necessary.

Comment (laurapugliese) [reply ts/boot/configureStoreAndPersistor.ts]: Fixed in commit 2bc1c30

File: ts/boot/configureStoreAndPersistor.ts (hunk 1 above)
Comment (fpersico) [reply ts/boot/configureStoreAndPersistor.ts]: Can you try something like.

```
const clonedState = {...state} as PersistentGlobalState;
delete clonedState.entities.organizations;
return clonedState;
```

Comment (fpersico) [reply ts/boot/configureStoreAndPersistor.ts]: What i mean is cast the state only once.

File: ts/boot/configureStoreAndPersistor.ts [hunk 4]
```diff
@@ -62,6 +62,20 @@ const migrations: MigrationManifest = {
   // Adds messagesIdsByServiceId
   "2": (state: PersistedState) => {
     return addMessagesIdsByServiceId(state as PersistedGlobalState);
+  },
+
+  // Version 3
+  // we changes the organizations so we clear the entitie of organizations to force a reload
+  "3": (state: PersistedState) => {
+    return {
```
[the remark this answers is not in the document]
Comment (fpersico) [reply ts/boot/configureStoreAndPersistor.ts]: If you want to really migrate you have to populate/calculate "all" too.

File: ts/boot/configureStoreAndPersistor.ts (hunk 1 above)
Comment (laurapugliese) [reply ts/boot/configureStoreAndPersistor.ts]: Ok! I try

Comment (laurapugliese) [reply ts/boot/configureStoreAndPersistor.ts]: @fpersico To use this code it is necessary to remove the Readonly attribute on EntitiesState and disable a tslint rule (no-object-mutation). What do you think? I proceed?

Comment (fpersico) [reply ts/boot/configureStoreAndPersistor.ts]: No. Just cast the state once. But please check what @Undermaken said: "I tested the migration but an error is still living".

**Comment by laurapugliese** [comment]
@fpersico @Undermaken In the commit a950517 I have added also the migration of all organizations (@fpersico don't kill me :) but I had to use `any` otherwise the organizations into entities get the new structure). I'm waiting for your feedback.

**Comment by Undermaken** [comment]
@fpersico could you review last changes?

**Pull request merged**